### PR TITLE
fix(docgen): add expanded type name when union uses in define-props ts annotation

### PR DESCRIPTION
**Pull request by avinean**
Fix displaying union types as 'union' instead of 'string | number | boolean'

For example if I have such component
```
  defineProps<{
      /**
     * Size of the button that determine font size and paddings size
     */
    size: 'xs' | 'sm' | 'md' | 'lg'
  }>()
```

I would expect to see this in args table
image

What I see instead
image

I assume PR will close this https://github.com/storybookjs/storybook/issues/16301

### CODE REVIEW

**Comment by elevatebart** [comment]
This change is a breaking change in the API. I will implement it, merge it, at the same time as I merge true TypeScript support. It will be in vue-docgen-api 5.0.

Working on it with @johnsoncodehk, the author of volar.

See more at
https://github.com/johnsoncodehk/volar/tree/master/packages/vue-component-meta

I wish I could say more about the timeline.

**Comment by elevatebart** [comment]
Let me rephrase that...

First, this is a great PR. 
There is tests, a great description and a very good use case. 
Thank you for that.

Unfortunately, we are not yet at a point where this can be merged in the `dev` branch and I have not started the `next` branch.

Sorry about that,
I will keep you posted.

**Review by elevatebart — APPROVED** [review]

**Comment by jnarowski** [comment]
I need this too. Without it the vue3+typescript setup is very hard to use. 

Any idea on timeline or work arounds in the meantime?

**Comment by thomasaull** [comment]
Dear stalebot, please don‘t close this issue automatically 🤝
